cfme/networks/topology.py

```diff
@@ -0,0 +1,222 @@
+import re
+import attr
+
+from navmazing import NavigateToAttribute
+
+from cfme.modeling.base import BaseCollection, BaseEntity
+from cfme.networks.topology_view import TopologyView
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from wait_for import wait_for
+
+
+@attr.s
+class TopologyLegend(BaseEntity):
+    element = attr.ib()
+    legend_name = attr.ib()
+
+    @property
+    def name(self):
+        return self.legend_name
+
+    @property
+    def is_active(self):
+        return 'active' in self.element.get_attribute('class')
+
+    def set_active(self, active=True):
+        if active != self.is_active:
+            self.element.click()
+
+
+@attr.s
+class TopologyLegendCollection(BaseCollection):
+    """Collection object for legends in topology"""
+    ENTITY = TopologyLegend
+
+    def all(self, topology):
```

Comment:
**REQUIRED**
It looks like this method filters out TopologyLegend by Topology. So it should called `filter`.